app/src/main/java/com/sun/noteapp/utils/ConvertString.kt

```diff
@@ -1,43 +1,45 @@
 package com.sun.noteapp.utils
 
-object ConvertString {
+import com.sun.noteapp.data.model.ToDo
 
-    fun actionListToActionDataString(actions: List<String>): String {
-        var result = ""
-        actions.forEach {
-            result += "_0$it"
-        }
-        result = result.substring(1)
-        return result
-    }
+object ConvertString {
 
-    fun actionDataStringToActionCheckList(str: String): List<Pair<String, Boolean>> {
-        val checkList = mutableListOf<Pair<String, Boolean>>()
-        val arrayS = str.split("_")
-        arrayS.forEach {
-            val check = it[0] == '1'
-            checkList.add(Pair(it.substring(1), check))
+    fun convertToDoDataStringToList(dataString: String): List<ToDo> {
+        val results = mutableListOf<ToDo>()
+        if (dataString.isNotEmpty()) {
+            val arrayToDo = dataString.split("$UNDER_STROKE")
+            arrayToDo.map {
+                val isComplete = it[0] == '1'
+                results.add(ToDo(it.substring(1), isComplete))
+            }
         }
-        return checkList
+        return results
```

Comment:
list.map{} sẽ tạo ra một list mới với số phần tử tương ứng với list cũ, nhưng mỗi phần tử của list mới sẽ được biến đổi từ phần tử của list cũ thông qua hàm biến đổi mà ta cài đặt. 
ví dụ: list cũ : [1 , 2, 3] 
biến đổi thông qua hàm it -> it*2 
ta sẽ được list mới [2, 4, 6]
Trong bài toán của anh, list cũ là dataString.split() 
list mới là 1 list ToDo 
```suggestion
        fun convertToDoDataStringToList(dataString: String): List<ToDo> = 
            if (dataString.isEmpty()){
                emptyList()
            } else {
                dataString.split("$UNDER_STROKE").map { item: String -> 
                    ToDo(item.subString(1), item[0] == '1')
                }
            }
```